subworkflows/local/prepare_genome/main.nf

```diff
@@ -100,6 +104,39 @@ workflow PREPARE_GENOME {
         versions = versions.mix(UNTAR_CHR_DIR.out.versions)
     } else chr_files = Channel.fromPath(chr_dir).collect()
 
+    // prepare genome for bbsplit
+    ch_bbsplit_index = Channel.empty()
+    if (!params.skip_bbsplit){
+        // Create a simple value channel for the fasta file
+        ch_fasta = fasta
+            .map { meta, path -> path }
+            .first()
+            .map { path -> file(path) }
+        //
+        // Uncompress BBSplit index or generate from scratch if required
+        //
+        ch_bbsplit_index = Channel.empty()
+        if (bbsplit_index) {
+            if (bbsplit_index.endsWith('.tar.gz')) {
+                ch_bbsplit_index = UNTAR_BBSPLIT_INDEX ( [ [:], bbsplit_index ] ).untar.map { it[1] }
+                versions         = versions.mix(UNTAR_BBSPLIT_INDEX.out.versions)
+            } else {
+                ch_bbsplit_index = Channel.value(file(bbsplit_index))
+            }
+        } else {
+            Channel
+                .from(file(bbsplit_fasta_list))
+                .splitCsv() // Read in 2 column csv file: short_name,path_to_fasta
```

Comment:
can you use the samplesheet validation from nf-schema here? It comes with a bunch of neat utility functions and we have a schema to validate the file before the workflow starts